src/measure_theory/function/intersectivity.lean

```diff
@@ -0,0 +1,365 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import measure_theory.function.lp_space
+import measure_theory.integral.set_integral
+import order.upper_lower.locally_finite
+
+/-!
+# Bergelson's intersectivity lemma
+
+This file proves the Bergelson intersectivity lemma.
+
+## References
+
+[Bergelson, *Sets of recurrence of `ℤᵐ`-actions and properties of sets of differences in
+`ℤᵐ`][bergelson1985]
+
+## TODO
+
+Restate the theorem using the upper density of a set of naturals, once we have it.
+
+Use the ergodic theorem to deduce the refinement of the Poincaré recurrence theorem proved by
+Bergelson.
+-/
+
+attribute [measurability] measurable_one
+
+section
+open filter function measure_theory set
+open_locale ennreal
+variables {α : Type*} [measurable_space α] {μ : measure α} {s N : set α} {f : α → ℝ} {r : ℝ}
+
+attribute [simp] integrable_const
+
+@[simp] lemma integral_indicator_one (hs : measurable_set s) :
+  ∫ a, s.indicator 1 a ∂μ = (μ s).to_real :=
+(integral_indicator_const 1 hs).trans $ mul_one _
+
+/-- **First moment method**. An integrable function is smaller than its mean on a set of positive
```

Comment:
This has already been merged in master, in a different file. Can you update the PR to take this into account?

Reply:
Yes, but only partly :( I need the `lintegral` version.

Reply:
#19199